apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -82,6 +98,29 @@ defmodule Aecore.Chain.Worker do
     Enum.reverse(get_blocks([], start_block_hash, final_block_hash, count))
   end
 
+  @spec longest_blocks_chain() :: list(Block.t())
+  def longest_blocks_chain() do
+    get_blocks(top_block_hash(), top_height() + 1)
+  end
+ 
+  def get_chain_state_by_height(height, chain_hash \\ nil) do
+    get_block_info_by_height(height, chain_hash).chain_state
+  end
+
+  @spec chain_state(binary()) :: ChainState.account_chainstate()
```

Comment:
what does this function give? Chain-State at height? If so, how is it computed efficiently?

Reply:
Chainstate at hash. It's not a new function.

Reply:
I didn't notice we already had this...